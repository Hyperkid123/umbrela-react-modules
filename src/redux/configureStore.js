import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
