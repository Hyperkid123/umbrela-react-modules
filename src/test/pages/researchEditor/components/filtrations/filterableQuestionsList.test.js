import React from 'react'
import FilterableQuestionsList, {FilterableQuestionsList as Snapshot} from '../../../../../pages/researchEditorModule/components/filtrations/filterableQuestionsList';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('FilterableQuestionsList component', () => {
    const initialState  = {
      options:{
        optionsFilters: {}
      },
      questions: {
        allQuestions:{
          1: [{
            questionId: 1,
            title: 'foo',
          }]
        }
      },
    };

    const mockStore = configureStore();
    it('should create component correctly', () => {
      let store = mockStore(initialState);
      let wrapper = shallow(<FilterableQuestionsList store={store}/>);
      expect(wrapper).toBeDefined();
    });

    it('should render correctly with questions', () => {
      const tree = shallow(
        <Snapshot
          loadQuestions={jest.fn()}
          questions={[{
            questionId: 1,
            title: 'foo',
          }]}
          filters={[{}]}
          selectedOptionId={0}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
    });
});
