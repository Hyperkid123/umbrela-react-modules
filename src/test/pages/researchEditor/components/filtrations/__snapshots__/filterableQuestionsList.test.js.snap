// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableQuestionsList component should render correctly with questions 1`] = `
<List
  classes={
    Object {
      "dense": "MuiList-dense-3",
      "padding": "MuiList-padding-2",
      "root": "MuiList-root-1",
      "subheader": "MuiList-subheader-4",
    }
  }
  component="ul"
  dense={false}
  disablePadding={false}
  style={
    Object {
      "margin": 5,
    }
  }
>
  <WithStyles(ListItem)>
    <WithStyles(ListItemIcon)>
      <pure(List) />
    </WithStyles(ListItemIcon)>
    <WithStyles(ListItemText) />
  </WithStyles(ListItem)>
  <WithStyles(Divider) />
  <WithStyles(ListItem)
    button={true}
    dense={true}
    key="1"
    onClick={[Function]}
  >
    <WithStyles(Checkbox)
      checked={false}
    />
    <WithStyles(ListItemText)
      primary="foo"
    />
  </WithStyles(ListItem)>
</List>
`;
